### PR TITLE
Mechanics can set access requirements of constructed airlocks

### PR DESCRIPTION
**Pull request by penny64**
[FEATURE][QOL]
## About the PR 

image
image
image
image

Constructed airlocks now spawn with no access requirements.
Mechanics can set the access requirements of newly constructed airlocks with the new Access Lite, found in their lockers.

It is able to scan existing airlocks and set the access requirements of constructed airlocks, but only once, any edits require getting the HOP just like before.

Restricted airlocks cannot be scanned.

I also adjust the name of the access-pro, for consistency.

## Why's this needed? 

Airlocks spawned with their "default" access list which half the time wasn't what you wanted, given access lists.
This makes constructing any airlock actually useful.

## Changelog

```changelog
(u)Penny
(*)Mechanics can now set the access requirements of newly constructed airlocks- find the new Access Lite in your secure lockers to try it out!
```

### CODE REVIEW

**Comment by nefarious6th** [comment]
Access Pro is niche enough and HoP's unique tool. All this does is devalue the unique role and tool of the HoP and make mechanics more isolationist versus having HoP or their own department head the CE help them with setting door access, since both can use the Pro with their ID levels.

**Comment by penny64** [comment]
Construction is unfun needing heads to build an airlock and the current behavior felt bad and unintended.
Overall the Access Pro wasn't made for construction, as it's slow and cumbersome to use and requires setting up ID cards with the access you want.
The Access Pro's use is for re-programming or removing doors, not construction.

**Comment by PrimeNumb** [comment]
I agree that requiring micromanagement from the HoP just to construct new airlocks is awkward at best.

**Comment by DimWhat** [comment]
While this would be good, this doesn't really ring to me as a "mechanic tool." Considering that a mechanic can just scan their own door and hit a random door in maint to make it locked to mechanics, I'm not not keen on it.

**Comment by penny64** [comment]
@DimWhat yep, can only program fresh, player constructed airlock!

**Comment by Awfullyfishy** [comment]
Our current options for making new airlocks is: 
CE's RCD as the ideal, as it can name, access and select skins for the airlocks.
Scanning the door skin you want and printing it, and calling HoP to fix the access. Not sure you can scan any new door skins, the Ruckingenur is buggy with multiple door scans I believe.
Using the normal RCD and crying in your ugly maint door.

It'd be nice if the Access Lite wasn't needed at all and Mechanics had the ability to specify requirements on printing a blueprint, but thats more work.

**Comment by penny64** [comment]
Ruckingenur scanning has been fixed in my recent PRs. and most doors have unique names!

File: code/obj/item/device/accessgun.dm [hunk 1]
```diff
@@ -174,3 +178,55 @@
...
+				boutput(user, "<span class='notice'>[src] has no access requirements loaded.</span>")
+				return
+			if (length(door_reqs.req_access))
+				playsound(src, 'sound/machines/airlock_deny.ogg', 35, 1, 0, 2)
+				boutput(user, "<span class='notice'>[src] cannot reprogram [door_reqs.name], access requirements already set.</span>")
+				return
+			. = ..()
+			return
+		if(is_restricted(door_reqs))
+			playsound(src, 'sound/machines/airlock_deny.ogg', 35, 1, 0, 2)
```
Comment (ZeWaka) [inline code/obj/item/device/accessgun.dm:211]:
```suggestion
				playsound(src, "sound/machines/airlock_deny.ogg", 35, 1, 0, 2)
				boutput(user, "<span class='notice'>[src] has no access requirements loaded.</span>")
				return
			if (length(door_reqs.req_access))
				playsound(src, "sound/machines/airlock_deny.ogg", 35, 1, 0, 2)
				boutput(user, "<span class='notice'>[src] cannot reprogram [door_reqs.name], access requirements already set.</span>")
				return
			. = ..()
			return
		if(is_restricted(door_reqs))
			playsound(src, "sound/machines/airlock_deny.ogg", 35, 1, 0, 2)
```

**Review by ZeWaka — APPROVED** [review]

File: code/obj/item/device/accessgun.dm (hunk 1 above)
Comment (ZeWaka) [inline code/obj/item/device/accessgun.dm:211]:
```suggestion
			playsound(src, "sound/machines/airlock_deny.ogg", 35, 1, 0, 2)
```

File: code/obj/item/device/accessgun.dm [hunk 2]
```diff
@@ -174,3 +178,55 @@
...
+		if (!istype(door_reqs))
+			. = ..()
+			return
+		if(target.deconstruct_flags & DECON_BUILT)
+			if (isnull(scanned_access))
+				playsound(src, 'sound/machines/airlock_deny.ogg', 35, 1, 0, 2)
+				boutput(user, "<span class='notice'>[src] has no access requirements loaded.</span>")
+				return
+			if (length(door_reqs.req_access))
+				playsound(src, 'sound/machines/airlock_deny.ogg', 35, 1, 0, 2)
```
Comment (ZeWaka) [inline code/obj/item/device/accessgun.dm:205]:
```suggestion
				playsound(src, "sound/machines/airlock_deny.ogg", 35, 1, 0, 2)
```

File: code/obj/item/device/accessgun.dm [hunk 3]
```diff
@@ -174,3 +178,55 @@
...
+		. = ..()
+
+	afterattack(obj/target, mob/user, reach, params)
+		var/obj/machinery/door/airlock/door_reqs = target
+		if (!istype(door_reqs))
+			. = ..()
+			return
+		if(target.deconstruct_flags & DECON_BUILT)
+			if (isnull(scanned_access))
+				playsound(src, 'sound/machines/airlock_deny.ogg', 35, 1, 0, 2)
```
Comment (ZeWaka) [inline code/obj/item/device/accessgun.dm:201]:
```suggestion
				playsound(src, "sound/machines/airlock_deny.ogg", 35, 1, 0, 2)
```

**Review by ZeWaka — APPROVED** [review]

**Comment by ZeWaka** [comment]
gib permissons

**Comment by penny64** [comment]
@ZeWaka given

**Pull request merged**